### PR TITLE
Fixes to pass NeXus validation

### PR DESCRIPTION
**Pull request by lukaspie**
Using internal representation of nxdl's as tree structure from https://github.com/FAIRmat-NFDI/pynxtools/pull/323.

The writing currently works for (with no errors thrown in the validation):
- SPECS data:
  - [x] .sle files, propietary format of SPECS GmbH
  - [x] .xml: SpecsLab 2files, XML format from SPECS GmbH, software version v.1.6,
  - [x] xy: SpecsLabProdigy export format in XY format

- [Phi MultiPak](https://www.phi.com/surface-analysis-equipment/genesis.html#software:multi-pak-data-reduction-software/) data:
  - [x] .spe: regular spectra
  - [x] .pro: sputter depth profiling

- [Scienta Omicron](https://scientaomicron.com/en) data:
  - [x] .ibw: Igor binary wave
  - [x] .txt: text file exports

- .vms: VAMAS files:
  - [x] REGULAR format
  - [x] IRREGULAR format
  - [x] txt export from CasaXPS (for some reason not working in the CI/CD test yet, will be fixed later -> #51)

### CODE REVIEW

File: tests/test_reader.py [hunk 1]
```diff
@@ -3,53 +3,55 @@
...
+    ("phi_spe", "Phi .spe reader"),
+    ("phi_pro", "Phi .pro reader"),
+    ("specs_sle", "SPECS .sle reader"),
+    ("specs_xml", "SPECS .xml reader"),
+    ("specs_xy", "SPECS .xy reader"),
+    ("scienta_ibw", "Scienta .ibw reader"),
+    ("scienta_txt", "Scienta .txt export reader"),
+    ("vms_irregular", "Irregular VAMAS reader"),
+    ("vms_regular", "Regular VAMAS reader"),
+    # ("vms_txt_export", "Vamas txt export"),
```
Comment (RubelMozumder) [inline tests/test_reader.py:36]: From you discord comment, I remember you have problem with vms file. Right?

Comment (RubelMozumder) [reply tests/test_reader.py:36]: If the reader does not work with the vamas file, you can deactivate the option for running vamas example. if it is issue with test but generating nexus file with out any issue you can keep as it is.

File: examples/vms/eln_data_vms.yaml [hunk 2]
```diff
@@ -44,10 +42,9 @@ instrument:
         identifier: null
     energydispersion:
       scheme: hemispherical
-      entrance_slit: unknown
       energy_scan_mode: fixed_analyser_transmission
       diameter:
-        unit: mm
+        unit: milli-m
```
Comment (RubelMozumder) [inline examples/vms/eln_data_vms.yaml:47]: is this compatible with nomad? Nomad accept `mm`.

File: pynxtools_xps/phi/spe_pro_phi.py [hunk 3]
```diff
@@ -238,23 +239,23 @@
...
-    "scan_deflection_offset_x": "mm",
-    "scan_deflection_offset_y": "mm",
-    "xray_stigmator_x": "mm",
-    "xray_stigmator_y": "mm",
+    "scan_deflection_span_x": "milli-m",
+    "scan_deflection_span_y": "milli-m",
+    "scan_deflection_offset_x": "milli-m",
+    "scan_deflection_offset_y": "milli-m",
+    "xray_stigmator_x": "milli-m",
+    "xray_stigmator_y": "milli-m",
```
Comment (RubelMozumder) [inline pynxtools_xps/phi/spe_pro_phi.py:258]: The same is here about the units.

File: tests/data/scienta_txt/eln_data_scienta_txt.yaml [hunk 4]
```diff
@@ -26,17 +23,17 @@ instrument:
...
     distance:
-      unit: mm
+      unit: milli-m
       value: 0.0
-    #current:
-      #value: 30
-      #unit: "pA"
+    current:
+      value: 30
+      unit: pico-A
```
Comment (RubelMozumder) [inline tests/data/scienta_txt/eln_data_scienta_txt.yaml:36]: Ensure the units with nomad.

**Review by RubelMozumder — APPROVED** [review]
I did not go very deep. As all the sub readers have been implement previous only fits them with the new validation stage.
LGTM!

File: tests/test_reader.py (hunk 1 above)
Comment (lukaspie) [reply tests/test_reader.py:36]: It does not work for this file, but I will have a fix coming in soon.

File: examples/vms/eln_data_vms.yaml (hunk 2 above)
Comment (lukaspie) [reply examples/vms/eln_data_vms.yaml:47]: So far, these units have worked. There will be unit validation soon, then I may refactor this part.

**Pull request merged**